scripts/irods/test/test_federation.py

```diff
@@ -1943,3 +1943,222 @@ def test_iget_data_object_as_user_with_null_access_and_replica_only_in_archive__
 			self.remote_admin.run_icommand(
 				['iadmin', 'modrepl', 'logical_path', logical_path, 'resource_hierarchy', archive_hierarchy, 'DATA_REPL_STATUS', '0'])
 			self.remote_admin.run_icommand(['irm', '-f', logical_path])
+
+
+class test_icp_overwrite_with_target_resource(unittest.TestCase):
+	@classmethod
+	def setUpClass(self):
+		# Create a session as the administrator for the remote zone so we can do things in the remote zone.
+		self.remote_admin = session.make_session_for_existing_user(
+			test.settings.PREEXISTING_ADMIN_PASSWORD,
+			test.settings.PREEXISTING_ADMIN_PASSWORD,
+			test.settings.FEDERATION.REMOTE_HOST,
+			test.settings.FEDERATION.REMOTE_ZONE)
```

Comment:
Is this correct?

Shouldn't the first argument be the username?

Reply:
Well, you'd think that, but there is no such thing as `PREEXISTING_ADMIN_PASSWORD` (currently). I can add that and update this where this situation occurs. I think I lifted this from another test I wrote before...

Reply:
Done

Reply:
Sure, but does it actually pass with that? Does it not care about the username?

Reply:
The username for the admin is `rods` and the password is `rods`, so this works as-is. I went ahead and added the username as a separate item should we change this in the future.

Reply:
Ohhhh. Well that explains it.

Having an actual variable for `PREEXISTING_ADMIN_USERNAME` makes sense, even if it's identical to the password.

That can happen later too. Up to you.